src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -63,6 +64,69 @@ public void Kill()
             }
         }
 
+        /// <summary>Stops the associated process immediately.</summary>
+        private void Stop()
+        {
+            EnsureState(State.HaveNonExitedId);
+            if (Interop.Sys.Kill(_processId, Interop.Sys.Signals.SIGSTOP) != 0)
+            {
+                throw new Win32Exception(); // to match behavior of Kill()
+            }
+        }
+
+        private IEnumerable<Exception> KillTree()
+        {
+            List<Exception> exceptions = new List<Exception>();
```

Comment:
Instead of creating an aggregated list of exceptions, an alternative would be to stop on the first Process that causes an issue.
It may be simpler for the user to deal with a single exception.

Reply:
I wonder if an optional argument could eventually be added to request this behavior (something like `throwOnFirstFailure: true`). 

The catch with doing it by default is that it leaves the consumer without an easy way to continue termination if what they really want is best-effort, try to terminate all descendants, regardless of whether exceptions are encountered. 

If termination dies on an exception partway through and the consumer wants execution to resume, they won't necessarily be able to re-call `Kill(entireProcessTree: true)` on the `Process` instance they started with because that instance may be dead. To resume, they'd have to figure out which process(es) is/are the orphan(s) left when termination died because of the exception, then call  `Kill(entireProcessTree: true)` on each of them. I don't think the consumer will have an easy way to find those orphans. 

This makes me hesitate with making _die on first exception_ the default behavior.